gui/wxpython/dbmgr/base.py

```diff
@@ -1018,7 +1018,7 @@ def DeletePage(self, layer):
         if layer not in self.layers:
             return False
 
-        GNotebook.DeleteNBPage(self, self.layers.index(layer))
+        GNotebook.DeletePage(self, self.layers.index(layer))
```

Comment:
Could you please comment on this more? The new code uses directly the method of the FlatNotebook, but why? Is something wrong about GNotebook's implementation? Wouldn't it be possibly better to adjust the code of this function to work with the GNotebook's version of DeletNBPage?

Reply:
`NotebookController` base class, `InsertPage` method  has 2 way how to insert page:

https://github.com/OSGeo/grass/blob/120f198fca22a564334d7721fabbe8c6e299bf6b/gui/wxpython/gui_core/widgets.py#L143-L154

1. Insert page with name param
2. Insert page **without name** param (this is preferred way in Attribute Table Manager code)
https://github.com/OSGeo/grass/blob/120f198fca22a564334d7721fabbe8c6e299bf6b/gui/wxpython/dbmgr/base.py#L1173-L1178

`NotebookController` base class, inside `DeletePage` method is called method `GetPageIndexByName` which not return page index (return -1) if page was inserted without name param.

https://github.com/OSGeo/grass/blob/120f198fca22a564334d7721fabbe8c6e299bf6b/gui/wxpython/gui_core/widgets.py#L156-L169

`FlatNotebookController class`, `GetPageIndexByName` method 

https://github.com/OSGeo/grass/blob/120f198fca22a564334d7721fabbe8c6e299bf6b/gui/wxpython/gui_core/widgets.py#L258-L264

`GNotebook` class method `DeleteNBPage` implementation allow delete page which was inserted with name param only.